engines/suppliers/app/models/suppliers/suppliers_finder.rb

```diff
@@ -0,0 +1,16 @@
+module Suppliers
+  class SuppliersFinder
```

Comment:
The idea is to have this only for suppliers. We'll add this abstraction wherever we need. Keep in mind that the different index actions need different things.

Let's keep it as a `SuppliersFinder` for now. If there needed to add a new filter, I added it here.